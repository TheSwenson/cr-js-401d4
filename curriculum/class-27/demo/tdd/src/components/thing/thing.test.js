import React from 'react';
import renderer from 'react-test-renderer';
import Thing from './thing';

describe('<Thing />', () => {
  it('can render initially in section.thing', () => {
    let app = shallow(<Thing />);
    expect(app.find('section.thing').exists()).toBe(true);
  });

  it('has flag: true initially', () => {
    let app = shallow(<Thing />);
    expect(app.state('flag')).toBe(true);

    expect(app.find('span.flag').text()).toBe('True');
  });

  it('can click button to toggle flag', () => {
    // Arrange
    let app = shallow(<Thing />);
    let button = app.find('button');
    expect(button.exists()).toBe(true);
    expect(button.text()).toBe('Toggle');

    // Act
    button.simulate('click');

    // Assert
    expect(app.state('flag')).toBe(false);
    expect(app.find('span.flag').text()).toBe('False');

    // Act
    button.simulate('click');

    // Assert
    expect(app.state('flag')).toBe(true);
    expect(app.find('span.flag').text()).toBe('True');
  });

  it('renders consistently', () => {
    const tree = renderer.create(<Thing />).toJSON();
    // console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});
