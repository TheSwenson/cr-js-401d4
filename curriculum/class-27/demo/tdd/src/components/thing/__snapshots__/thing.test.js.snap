// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thing /> renders consistently 1`] = `
<section
  className="thing"
>
  <span
    className="flag"
  >
    True
  </span>
  <button
    onClick={[Function]}
  >
    Toggle
  </button>
</section>
`;
